src/main.nr

```diff
@@ -7,7 +9,15 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
```

Comment:
Were you able to get the tests working? I was running into issues, fixed your code to this to make it work

```
fn calculate_hash(num: u32) -> u32 {
    let size: u64 = 4294967296;
    let mut hash: u64 = 2166136261;

    let bytes = (num as Field).to_be_bytes(4);

    for idx in 0..4 {
        let product = (hash * PRIME as u64) % size;
        hash = product ^ bytes[idx] as u64;
    }
    hash as u32
}
```